build-logic/convention/src/main/kotlin/HiltConventionPlugin.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+import com.google.samples.apps.nowinandroid.libs
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.artifacts.VersionCatalog
+import org.gradle.kotlin.dsl.DependencyHandlerScope
+import org.gradle.kotlin.dsl.dependencies
+
+abstract class HiltConventionPlugin : Plugin<Project> {
```

Comment:
Could this be implemented as a delegate instead of inheriting?

Something like:

```
class HiltConventionPlugin(
    val basePluginId: String? = null, 
    dependencyHandler: DependencyHandlerScope.(libs: VersionCatalog) -> Unit
): Plugin<Project> {
 
    override fun apply(target: Project) {
        with(target) {
            with(pluginManager) {
                basePluginId?.let(::apply)
                // KAPT must go last to avoid build warnings.
                // See: https://stackoverflow.com/questions/70550883/warning-the-following-options-were-not-recognized-by-any-processor-dagger-f
                apply("org.jetbrains.kotlin.kapt")
            }
            dependencies {
                dependencyHandler()
            }
        }
    }
}
```